providers/flagd/pkg/provider.go

```diff
@@ -1,213 +1,164 @@
-			if ok := p.withCache(cacheValue); !ok {
-				p.logger.Error(fmt.Errorf("%s is invalid", cacheValue),
-					fmt.Sprintf("invalid env config for %s provided, using default value: %s",
-						flagdCacheEnvironmentVariableName, defaultCache,
-					))
-			}
-		}
+func (p *Provider) FloatEvaluation(
+	ctx context.Context, flagKey string, defaultValue float64, evalCtx of.FlattenedContext,
+) of.FloatResolutionDetail {
+	return p.service.ResolveFloat(ctx, flagKey, defaultValue, evalCtx)
+}
 
-		maxEventStreamRetriesS := os.Getenv(flagdMaxEventStreamRetriesEnvironmentVariableName)
-		if maxEventStreamRetriesS != "" {
-			maxEventStreamRetries, err := strconv.Atoi(maxEventStreamRetriesS)
-			if err != nil {
-				p.logger.Error(err,
-					fmt.Sprintf("invalid env config for %s provided, using default value: %d",
-						flagdMaxEventStreamRetriesEnvironmentVariableName, defaultMaxEventStreamRetries))
-			} else {
-				p.eventStreamConnectionMaxAttempts = maxEventStreamRetries
-			}
-		}
-	}
+func (p *Provider) IntEvaluation(
+	ctx context.Context, flagKey string, defaultValue int64, evalCtx of.FlattenedContext,
+) of.IntResolutionDetail {
+	return p.service.ResolveInt(ctx, flagKey, defaultValue, evalCtx)
 }
 
-func (p *Provider) withCache(cache string) bool {
-	switch cache {
-	case cacheDisabledValue:
-		WithoutCache()(p)
-	case cacheLRUValue:
-		WithLRUCache(p.maxCacheSize)(p)
-	default:
-		return false
-	}
+func (p *Provider) ObjectEvaluation(
+	ctx context.Context, flagKey string, defaultValue interface{}, evalCtx of.FlattenedContext,
+) of.InterfaceResolutionDetail {
+	return p.service.ResolveObject(ctx, flagKey, defaultValue, evalCtx)
+}
+
+// ProviderOptions
+
+type ProviderOption func(*Provider)
 
-	return true
+// WithSocketPath overrides the default hostname and expectPort, a unix socket connection is made to flagd instead
+func WithSocketPath(socketPath string) ProviderOption {
+	return func(p *Provider) {
+		p.providerConfiguration.SocketPath = socketPath
+	}
 }
 
-// WithCertificatePath specifies the location of the certificate to be used in the gRPC dial credentials. If certificate loading fails insecure credentials will be used instead
+// WithCertificatePath specifies the location of the certificate to be used in the gRPC dial credentials.
+// If certificate loading fails insecure credentials will be used instead
 func WithCertificatePath(path string) ProviderOption {
 	return func(p *Provider) {
 		p.providerConfiguration.CertificatePath = path
 		p.providerConfiguration.TLSEnabled = true
 	}
 }
 
-// WithPort specifies the port of the flagd server. Defaults to 8013
+// WithPort specifies the expectPort of the flagd server. Defaults to 8013
 func WithPort(port uint16) ProviderOption {
 	return func(p *Provider) {
 		p.providerConfiguration.Port = port
 	}
 }
 
-// WithHost specifies the host name of the flagd server. Defaults to localhost
+// WithHost specifies the expectHost name of the flagd server. Defaults to localhost
```

Comment:
```suggestion
// WithHost specifies the host name of the flagd server. Defaults to localhost
```
guessing this is a typo

Reply:
🤦 this is from a test variable renaming. Thank you for noticing